handlers/productsHandler.js

```diff
@@ -0,0 +1,57 @@
+const Product = require('../model/productsModel');
+
+
+const allProducts = async (req,res) => {
+	const productsList = await Product.find();
+
+	if(!productsList) {
+		return res.status(500).json({success:false})
+	}
+	res.status(200).send(productsList);
+}
+
+// get all products posted by a specific user
+const productsByUserId = async (req,res) => {
+	const userId = req.params.userId;
+
+	try {
+		const productsById = await Product.find({user:userId});
+		res.status(200).json(productsById)
+	} catch (error) {
+		res.status(500).json({error:'failed to fetch products'})
+	}
+}
+
+
+// posting products by a specific user
+// const productsCreation = async , authUser(req,res) => {
+// 	const userId = req.user._id
+// }
+
+const productsCreation = async(req,res) => {
+	const userId = req.user._id;
+
+	try {
+		
+		const product = new Product({
+			name: req.body.name,
+			description: req.body.description,
+			price: req.body.price,
+			user: userId,
+		});
+		console.log(product)
+		product
+		.save()
```

Comment:
```suggestion
		const product = await product.save();
		res.status(200).json(product);
```
is this working at all ? so when you create a product you don't return anything ? please relate your reasons.